### ISSUE TITLE
Python error when loading python plugins with pyside binding

### ISSUE THREAD

**Issue by meyerj** [ros-visualization/rqt#91]
I currently get the following error message when a python plugin is loaded in rqt and pyside bindings are selected:

```
$ rqt -b pyside
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/python2.7/dist-packages/qt_gui/main_window.py", line 91, in restore_state
    self._restore_state_from_perspective()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/qt_gui/main_window.py", line 129, in _restore_state_from_perspective
    self.addToolBar(area, toolbar)
TypeError: 'PySide.QtGui.QMainWindow.addToolBar' called with wrong argument types:
  PySide.QtGui.QMainWindow.addToolBar(int, MinimizedDockWidgetsToolbar)
Supported signatures:
  PySide.QtGui.QMainWindow.addToolBar(PySide.QtGui.QToolBar)
  PySide.QtGui.QMainWindow.addToolBar(PySide.QtCore.Qt.ToolBarArea, PySide.QtGui.QToolBar)
  PySide.QtGui.QMainWindow.addToolBar(unicode)
```

I have not observed any side effects yet, but I guess most plugins do not have any toolbars.

My system is Ubuntu precise amd64 with ROS hydro and rqt installed from the official repositories.

Versions:

```
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                 Version                                              Description
+++-====================================================-====================================================-========================================================================================================================
ii  ros-hydro-rqt-action                                 0.3.2-0precise-20131016-0054-+0000                   rqt_action provides a feature to introspect all available ROS action (from actionlib) types. By utilizing rqt_msg, the o
ii  ros-hydro-rqt-bag                                    0.3.2-0precise-20131015-2216-+0000                   rqt_bag provides a GUI plugin for displaying and replaying ROS bag files.
ii  ros-hydro-rqt-bag-plugins                            0.3.2-0precise-20131015-2230-+0000                   rqt_bag provides a GUI plugin for displaying and replaying ROS bag files.
ii  ros-hydro-rqt-common-plugins                         0.3.2-0precise-20131021-0153-+0000                   rqt_common_plugins metapackage provides ROS backend graphical tools suite that can be used on/off of robot runtime. To r
ii  ros-hydro-rqt-console                                0.3.2-0precise-20131015-2308-+0000                   rqt_console provides a GUI plugin for displaying and filtering ROS messages.
ii  ros-hydro-rqt-dep                                    0.3.2-0precise-20131015-2308-+0000                   rqt_dep provides a GUI plugin for visualizing the ROS dependency graph.
ii  ros-hydro-rqt-graph                                  0.3.2-0precise-20131015-2244-+0000                   rqt_graph provides a GUI plugin for visualizing the ROS computation graph. Its components are made generic so that other
ii  ros-hydro-rqt-gui                                    0.2.12-0precise-20131015-1954-+0000                  rqt_gui provides the main to start an instance of the ROS integrated graphical user interface provided by qt_gui.
ii  ros-hydro-rqt-gui-cpp                                0.2.12-0precise-20131015-2054-+0000                  rqt_gui_cpp enables GUI plugins to use the C++ client library for ROS.
ii  ros-hydro-rqt-gui-py                                 0.2.12-0precise-20131015-2005-+0000                  rqt_gui_py enables GUI plugins to use the Python client library for ROS.
ii  ros-hydro-rqt-image-view                             0.3.2-0precise-20131020-2349-+0000                   rqt_image_view provides a GUI plugin for displaying images using image_transport.
ii  ros-hydro-rqt-launch                                 0.3.2-0precise-20131016-0014-+0000                   This rqt plugin ROS package provides easy view of .launch files. User can also start and end node by node that are defin
ii  ros-hydro-rqt-logger-level                           0.3.2-0precise-20131015-2243-+0000                   rqt_logger_level provides a GUI plugin for configuring the logger level of ROS nodes. rqt_logger_level takes over `wx`-b
ii  ros-hydro-rqt-moveit                                 0.3.1-0precise-20131015-2244-+0000                   An rqt-based tool that assists monitoring tasks for MoveIt! motion planner developers and users. Currently the following
ii  ros-hydro-rqt-msg                                    0.3.2-0precise-20131016-0013-+0000                   A Python GUI plugin for introspecting available ROS message types. Note that the msgs available through this plugin is t
ii  ros-hydro-rqt-nav-view                               0.3.1-0precise-20131016-0059-+0000                   rqt_nav_view provides a gui for viewing navigation maps and paths.
ii  ros-hydro-rqt-plot                                   0.3.2-0precise-20131015-2244-+0000                   rqt_plot provides a GUI plugin visualizing numeric values in a 2D plot using different plotting backends.
ii  ros-hydro-rqt-pose-view                              0.3.1-0precise-20131016-0059-+0000                   rqt_pose_view provides a GUI plugin for visualizing 3D poses.
ii  ros-hydro-rqt-publisher                              0.3.2-0precise-20131015-2244-+0000                   rqt_publisher provides a GUI plugin for publishing arbitrary messages with fixed or computed field values.
ii  ros-hydro-rqt-py-common                              0.3.2-0precise-20131015-2230-+0000                   rqt_py_common provides common functionality for rqt plugins written in Python. Despite no plugin is provided, this packa
ii  ros-hydro-rqt-py-console                             0.3.2-0precise-20131015-2027-+0000                   rqt_py_console is a Python GUI plugin providing an interactive Python console.
ii  ros-hydro-rqt-reconfigure                            0.3.2-0precise-20131016-0013-+0000                   This rqt plugin succeeds former dynamic_reconfigure's GUI (reconfigure_gui), and provides the way to view and edit the p
ii  ros-hydro-rqt-robot-dashboard                        0.3.1-0precise-20131016-0135-+0000                   rqt_robot_dashboard provides an infrastructure for building robot dashboard plugins in rqt.
ii  ros-hydro-rqt-robot-monitor                          0.3.1-0precise-20131015-2307-+0000                   rqt_robot_monitor displays diagnostics_agg topics messages that are published by diagnostic_aggregator. rqt_robot_monito
ii  ros-hydro-rqt-robot-plugins                          0.3.1-0precise-20131021-0341-+0000                   Metapackage of rqt plugins that are particularly used with robots during its operation. To run any rqt plugins, just typ
ii  ros-hydro-rqt-robot-steering                         0.3.1-0precise-20131015-2230-+0000                   rqt_robot_steering provides a GUI plugin for steering a robot using Twist messages.
ii  ros-hydro-rqt-runtime-monitor                        0.3.1-0precise-20131015-2027-+0000                   rqt_runtime_monitor provides a GUI plugin viewing DiagnosticsArray messages.
ii  ros-hydro-rqt-rviz                                   0.3.1-0precise-20131021-0156-+0000                   rqt_rviz provides a GUI plugin embedding RViz. Note that this rqt plugin does NOT supersede RViz but depends on it.
ii  ros-hydro-rqt-service-caller                         0.3.2-0precise-20131015-2243-+0000                   rqt_service_caller provides a GUI plugin for calling arbitrary services.
ii  ros-hydro-rqt-shell                                  0.3.2-0precise-20131015-2027-+0000                   rqt_shell is a Python GUI plugin providing an interactive shell.
ii  ros-hydro-rqt-srv                                    0.3.2-0precise-20131016-0055-+0000                   A Python GUI plugin for introspecting available ROS message types. Note that the srvs available through this plugin is t
ii  ros-hydro-rqt-tf-tree                                0.3.1-0precise-20131015-2308-+0000                   rqt_tf_tree provides a GUI plugin for visualizing the ROS TF frame tree.
ii  ros-hydro-rqt-top                                    0.3.2-0precise-20131015-2027-+0000                   RQT plugin for monitoring ROS processes.
ii  ros-hydro-rqt-topic                                  0.3.2-0precise-20131015-2230-+0000                   rqt_topic provides a GUI plugin for displaying debug information about ROS topics including publishers, subscribers, pub
ii  ros-hydro-rqt-web                                    0.3.2-0precise-20131015-2027-+0000                   rqt_web is a simple web content viewer for rqt. Users can show web content in Qt-based window by specifying its URL.
ii  python                                               2.7.3-0ubuntu2.2                                     interactive high-level object-oriented language (default version)
ii  python-pyside                                        1.1.0-1                                              Python bindings for Qt4 (big metapackage)
```

**Comment by DorianScholz** [comment]
I had some problems with PySide 1.1.0 as well on other projects and installed 1.1.2 from their official PPA:
ppa:pyside/ppa
Could you try if this fixes it for you?

**Comment by meyerj** [comment]
Nope, still the same error.
Do I have to clean some caches or run rqt from a source build? My /opt/ros/hydro/lib/python2.7/dist-packages directory still contains lots of `*.pyc` files.

Updated versions:

```
ii  libpyside-dev                              1.1.2-1~p0                                          Python bindings for Qt 4 (development files)
ii  libpyside-py3-1.1                          1.1.2-1~p0                                          Python3 bindings for Qt 4 (base files)
ii  libpyside1.1                               1.1.2-1~p0                                          Python bindings for Qt 4 (base files)
ii  python-pyside                              1.1.2-1~p0                                          Python bindings for Qt4 (big metapackage)
ii  python-pyside.phonon                       1.1.2-1~p0                                          Qt 4 Phonon module - Python bindings
ii  python-pyside.qtcore                       1.1.2-1~p0                                          Qt 4 core module - Python bindings
ii  python-pyside.qtdeclarative                1.1.2-1~p0                                          Qt 4 Declarative module - Python bindings
ii  python-pyside.qtgui                        1.1.2-1~p0                                          Qt 4 GUI module - Python bindings
ii  python-pyside.qthelp                       1.1.2-1~p0                                          Qt 4 help module - Python bindings
ii  python-pyside.qtnetwork                    1.1.2-1~p0                                          Qt 4 network module - Python bindings
ii  python-pyside.qtopengl                     1.1.2-1~p0                                          Qt 4 OpenGL module - Python bindings
ii  python-pyside.qtscript                     1.1.2-1~p0                                          Qt 4 script module - Python bindings
ii  python-pyside.qtsql                        1.1.2-1~p0                                          Qt 4 SQL module - Python bindings
ii  python-pyside.qtsvg                        1.1.2-1~p0                                          Qt 4 SVG module - Python bindings
ii  python-pyside.qttest                       1.1.2-1~p0                                          Qt 4 test module - Python bindings
ii  python-pyside.qtuitools                    1.1.2-1~p0                                          Qt 4 UI tools module - Python bindings
ii  python-pyside.qtwebkit                     1.1.2-1~p0                                          Qt 4 WebKit module - Python bindings
ii  python-pyside.qtxml                        1.1.2-1~p0                                          Qt 4 XML module - Python bindings
```

**Comment by DorianScholz** [comment]
To reproduce the bug I had to start rqt like this:
$ rqt --clear-config -b pyside
and then open a container from the plugin menu.
I fixed it in https://github.com/ros-visualization/qt_gui_core/commit/df70ef76f6a86682e3394fc163f9cb0309ec0d7b
Thanks for the report!

**Comment by meyerj** [comment]
Sorry for reporting so late, but the problem is still not completely solved with the current version 0.2.22 released in hydro. If there is a config from previous runs I get the following error message:

```
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/python2.7/dist-packages/qt_gui/main_window.py", line 91, in restore_state
    self._restore_state_from_perspective()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/qt_gui/main_window.py", line 127, in _restore_state_from_perspective
    area = Qt.ToolBarArea(toolbar_settings.value(toolbar.objectName(), Qt.NoToolBarArea))
TypeError: an integer is required
```

With `--clear-config` no error message is printed.

To reproduce it is sufficient to run rqt with `rqt --clear-config -b pyside`, then open a container, close the rqt instance and run again with `rqt -b pyside`.

**Comment by dirk-thomas** [comment]
Since the bug was fixed in 0.2.23 (https://github.com/ros-visualization/qt_gui_core/blob/groovy-devel/qt_gui/CHANGELOG.rst#0223-2014-05-07) it is obviously still there in 0.2.22.

The next patch release for Hydro will address this.
